bbbackend/models/user-services.js

```diff
@@ -45,63 +52,73 @@ async function getUsers(username, password) {
   return result;
 }
 
+// function to add a new user to a database based on the username and password that was input
+// sets defaults like the profile picture using the setDefaults function
+
 async function addUser(user) {
   try {
     const userToAdd = new userModel(user);
-    /*   const inDatabase = await findUserByNameAndPassword(
-      userToAdd.username,
-      userToAdd.password
-    );
-     
-    if (Object.keys(inDatabase).length === 0) {
-      setdefaults(userToAdd);
-      const savedUser = await userToAdd.save();
-      return savedUser;
-    }
-    */
-    setdefaults(userToAdd);
+    setDefaults(userToAdd);
     const savedUser = await userToAdd.save();
     return savedUser;
   } catch (error) {
     return false;
   }
 }
-function setdefaults(userToAdd) {
-  userToAdd.lists = { listcount: 0 };
-  userToAdd.friends = { friendcount: 0 };
-  userToAdd.groups = { testcount: 0 };
+
+
+// called in the addUser function, gives the new user an empty list for friends, groups, inventory, history
+
```

Comment:
the itemList attribute of a user is going to be a List of json objects of type "item" that each have the attributes itemName, price, quantity, and user. I believe you already updated the schema to reflect that, but we may have neglected the quantity attribute.